openmls/src/binary_tree/test_binary_tree.rs

```diff
@@ -0,0 +1,245 @@
+use crate::binary_tree::NodeIndex;
+
+use super::{array_representation::*, FLBBinaryTree};
+
+#[test]
+fn test_tree_creation() {
+    // Test tree creation: Wrong number of nodes.
+    let mut nodes = vec![0, 1];
+    assert_eq!(
+        ABinaryTree::new(nodes.clone())
+            .expect_err("No error when creating a non-full binary tree."),
+        ABinaryTreeError::InvalidNumberOfNodes
+    );
+    nodes.push(2);
+
+    // Test tree creation: Positive case.
+    let tree1 = ABinaryTree::new(nodes).expect("Error when creating tree from nodes.");
+    let mut tree2 = ABinaryTree::new(vec![0]).expect("Error when creating a one-node binary tree.");
+    tree2
+        .add(1, 2)
+        .expect("error when adding nodes to small enough tree");
+    assert_eq!(tree1, tree2);
+
+    // Test size reporting
+    assert_eq!(tree1.size(), 3);
+    assert_eq!(tree1.leaf_count(), 2);
+
+    // Test tree creation: Too many nodes.
+    unsafe {
+        let len = NodeIndex::max_value() as usize + 2;
+        let mut nodes: Vec<u32> = Vec::new();
+        nodes.set_len(len);
```

Comment:
Only this is unsafe really. But I guess it's fine to keep the entire "test" in an unsafe block.